src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/PrisonerProfileService.kt

```diff
@@ -1,52 +1,90 @@
 package uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.service
 
+import jakarta.validation.ValidationException
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.stereotype.Service
 import reactor.core.publisher.Mono
 import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.client.PrisonApiClient
 import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.client.PrisonerOffenderSearchClient
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.client.VisitSchedulerClient
 import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.PrisonerProfileDto
 import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.exception.InvalidPrisonerProfileException
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.filter.VisitSearchRequestFilter
 import java.time.Duration
+import java.time.LocalDateTime
+import java.time.Period
+import java.time.temporal.ChronoUnit
+import java.time.temporal.TemporalAdjusters
 
 @Service
 class PrisonerProfileService(
   private val prisonApiClient: PrisonApiClient,
   private val prisonerOffenderSearchClient: PrisonerOffenderSearchClient,
+  private val visitSchedulerClient: VisitSchedulerClient,
   @Value("\${prisoner.profile.timeout:10s}") private val apiTimeout: Duration,
-
+  @Value("\${prisoner.profile.past-visits.duration-in-months: P3M}") private val pastVisitsPeriod: Period,
+  @Value("\${prisoner.profile.future-visits.duration-in-months: P2M}") private val futureVisitsPeriod: Period,
 ) {
   companion object {
     val LOG: Logger = LoggerFactory.getLogger(this::class.java)
+    const val MAX_RECORDS: Int = 1000
+    const val PAGE: Int = 0
   }
 
   fun getPrisonerProfile(prisonId: String, prisonerId: String): PrisonerProfileDto? {
     val prisonerPromise = prisonerOffenderSearchClient.getPrisonerById(prisonerId)
```

Comment:
I know I said a promise in another review but I suppose they are Futures..

Reply:
Updated the variable names and also as discussed moved the logic from PrisonerProfileService to a new PrisonerProfileClient